### PR TITLE
Fixes storage eating buckled players

### LINKED ISSUE #9018: Closing a crate on yourself while riding a vehicle teleports you to arrivals
It also hides your sprite while you're in the vehicle but still shows the vehicle ontop of the crate
image

**Reproduction**
1. Get on a vehicle
2. Move ontop of a crate
3. Spam open and close the crate while moving slightly

**Video**

https://user-images.githubusercontent.com/31836813/174986770-115efbaf-2577-4408-9e81-ac2171a41584.mp4

### PR DESCRIPTION
**Pull request by Mirino97**
## About the PR 
Proposed fix #9018

This basically just adds a CancellableEvent to EntityStorageComponent and adds the Subscriptions to all components that were previously checked on manually inside EntityStorageComponent. Seems like an acceptable solution untill someone ECS's it entirely.

[WARNING]
I have no idea what I'm doing. Being so, please exagerate all scrutiny at your convenience. 
This bug seemed like a good way to learn more about the project and coding in general and this does look like a good solution to me, but feel free to close it in case it just isn't. Was a great learning experience :D

**Screenshots**
Video on YT cuz fuck changing my recording settings.
https://youtu.be/IFHnfUS87-U

**Changelog**

:cl: Sammy Boy
- fix: Crates cannot eat you while buckled anymore!

### CODE REVIEW

File: Content.Server/Ghost/GhostSystem.cs [hunk 1]
```diff
@@ -265,5 +267,11 @@ private IEnumerable<string> GetLocationNames()
 
             return players;
         }
+
+        public void OnStorageInsertAttempt(EntityUid uid, GhostComponent comp, InsertIntoStorageAttemptEvent args)
+        {
+            if (TryComp<GhostComponent>(uid, out var ghostcomp))
+                args.Cancel();
```
Comment (mirrorcult) [inline Content.Server/Ghost/GhostSystem.cs:274]:
```suggestion
            args.Cancel();
```

not sure if this one is even really necessary. they aren't items

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Server/Ghost/GhostSystem.cs (hunk 1 above)
Comment (Mirino97) [reply Content.Server/Ghost/GhostSystem.cs:274]: The check is necessary, i.e crates eat ghosts if this doesn't exist :3
But ye, I can remove the check for the component and it works a-ok!

**Review by mirrorcult — APPROVED** [review]

File: Content.Server/Storage/Components/EntityStorageComponent.cs [hunk 2]
```diff
@@ -365,6 +364,10 @@ protected virtual IEnumerable<EntityUid> DetermineCollidingEntities()
         }
     }
 
+    public sealed class InsertIntoStorageAttemptEvent : CancellableEntityEventArgs
+    {
+
+    }
```
Comment (mirrorcult) [inline Content.Server/Storage/Components/EntityStorageComponent.cs:370]: actually rename this to `InsertIntoEntityStorageAttemptEvent` cause i got confused by that and assumed this was for regular storage earlier

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Server/Storage/Components/EntityStorageComponent.cs (hunk 2 above)
Comment (Mirino97) [reply Content.Server/Storage/Components/EntityStorageComponent.cs:370]: You got it boss!

**Pull request merged**